scipy/stats/_distribution_infrastructure.py

```diff
@@ -1571,6 +1571,12 @@ def _update_parameters(self, *, validation_policy=None, **params):
         self._parameters = parameters
         self._parameterization = parameterization
         self._original_parameters = original_parameters
+        for name in self._parameters.keys():
+            # Make parameters properties of the class; return values from the instance
+            if hasattr(self.__class__, name):
+                continue
+            setattr(self.__class__, name, property(lambda self_, name_=name:
+                                                   self_._parameters[name_].copy()[()]))
```

Comment:
It would be nice if we could make the copies returned by the `fget` in the properties read only, so if someone tries to mutate them, they'll get immediate feedback they did something wrong, instead of having it silently do something they don't expect

we could do something like defining

```python
def _make_read_only_copy(x):
    out = x.copy()
    out.setflags(write=False)
    return out[()]
```

and then have

```python
setattr(self.__class__, name, property(lambda self_, name_=name:
                                                   _make_read_only_copy(self_._parameters[name_]))
```

Reply:
I considered this, but this infrastructure is soon(ish) going to work for any array backend, and read-only is not an array API thing.

Reply:
I see. And PyTorch doesn't even have read-only tensors, there's an open feature request for them here: https://github.com/pytorch/pytorch/issues/44027.